### ISSUE TITLE
REPL panics related to invalid unicode

### ISSUE THREAD

**Issue by solson** [denoland/deno#7953]
I found a way to trip two different Rust panic messages in the REPL:

```console
$ deno --version
deno 1.4.6
v8 8.7.220.3
typescript 4.0.3

$ deno
Deno 1.4.6
exit using ctrl+d or close()
> '\u{1f3b5}'
"🎵"
> '\u{1f3b5}'[0]
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error("unexpected end of hex escape", line: 1, column: 60)', cli/inspector.rs:859:54
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
fatal runtime error: failed to initiate panic, error 5
fish: “deno” terminated by signal SIGABRT (Abort)

$ deno
Deno 1.4.6
exit using ctrl+d or close()
> '\u{1f3b5}'[1]
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error("lone leading surrogate in hex escape", line: 1, column: 59)', cli/inspector.rs:859:54
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
fatal runtime error: failed to initiate panic, error 5
fish: “deno” terminated by signal SIGABRT (Abort)
```

Note that these panics *only* occur in the REPL, and regular Deno code is unaffected as far as I can tell.

cc @caspervonb

**Comment by caspervonb** [comment]
Thanks for the heads up 👍 

For reference, It's from a StringBuffer unwrap in the inspector so it only happens for local inspector messages that contains invalid unicode.

https://github.com/denoland/deno/blob/1956cb81372b96bc476e74ab43a62a6e60861277/cli/inspector.rs#L854-L868

**Comment by benjamingr** [comment]
Hey, I want to try and take a look at this as it's a relatively small fix that will make it learn some rust :]

**Comment by benjamingr** [comment]
Checked this for a bit, I think this actually isn't actually a small fix at all to do _well_. The "invalid" JSON is coming from the CDP itself, Chrome's JSON parser and other parsers I've found happily parse this and it seems like an issue with serde_json itself.

I will however take the opportunity to brush up on rust error handling and match expressions and try to solve the issue ad-hoc :]

**Comment by bartlomieju** [comment]
Fixed in #8759
